sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -368,190 +560,337 @@ public virtual PartitionProperties CreatePartitionPropertiesFromResponse(AmqpMes
         ///
         private static EventData BuildEventFromAmqpMessage(AmqpMessage source)
         {
-            var body = (source.BodyType.HasFlag(SectionFlag.Data))
-                ? ReadAmqpDataBody(source.DataBody)
-                : new BinaryData(ReadOnlyMemory<byte>.Empty);
+            var message = source switch
+            {
+                _ when TryGetDataBody(source, out var dataBody) => new AmqpAnnotatedMessage(dataBody),
+                _ when TryGetSequenceBody(source, out var sequenceBody) => new AmqpAnnotatedMessage(sequenceBody),
+                _ when TryGetValueBody(source, out var valueBody) => new AmqpAnnotatedMessage(valueBody),
+                _ => new AmqpAnnotatedMessage(AmqpMessageBody.FromData(MessageBody.FromReadOnlyMemorySegment(ReadOnlyMemory<byte>.Empty)))
+            };
 
-            ParsedAnnotations systemAnnotations = ParseSystemAnnotations(source);
+            // Header
 
-            // If there were application properties associated with the message, translate them
-            // to the event.
+            if ((source.Sections & SectionFlag.Header) > 0)
+            {
+                if (source.Header.DeliveryCount.HasValue)
```

Comment:
Some properties in the AMQP message can have a populated section with nothing but empty values present.  Because the `AmqpAnnotatedMessage` sections are lazily allocated when the accessor is invoked, each AMQP property value is being individually checked here.